Modules/_asynciomodule.c

```diff
@@ -2045,12 +2045,22 @@ static PyObject *
 swap_current_task(asyncio_state *state, PyObject *loop, PyObject *task)
 {
     PyObject *prev_task;
+
+    if (task == Py_None) {
+        if (PyDict_Pop(state->current_tasks, loop, &prev_task) < 0) {
+            return NULL;
+        }
+        if (prev_task == NULL) {
+            Py_RETURN_NONE;
+        }
+        return prev_task;
+    }
```

Comment:
It took me a minute to make sense of your changes in this file, but now I get it. 👍

Reply:
Same here. I blocked here for 5 minutes :-D